src/hugr/typecheck.rs

```diff
@@ -125,9 +125,26 @@ pub fn typecheck_const(typ: &ClassicType, val: &ConstValue) -> Result<(), ConstT
             (Container::Sum(_), _) => Err(ConstTypeError::Failed(ty.clone())),
             _ => Err(ConstTypeError::Unimplemented(ty.clone())),
         },
+        (ty @ ClassicType::Hashable(HashableType::Container(c)), tm) => match c {
+            // Here we deliberately build malformed Container-of-Hashable types
+            // (rather than Hashable-of-Container) in order to reuse logic above
```

Comment:
This is a bit of a code smell :-(, and hints that maybe there is a better way to do this. I think there is, but it's quite a big refactor, which I'll try to put up after this, as it's not urgent.